core/src/main/java/org/apache/calcite/sql/fun/SqlBasicAggFunction.java

```diff
@@ -181,7 +184,21 @@ public SqlBasicAggFunction withAllowsSeparator(boolean allowsSeparator) {
         getReturnTypeInference(), getOperandTypeInference(),
         getOperandTypeChecker(), getFunctionType(), requiresOrder(),
         requiresOver(),  requiresGroupOrder(), distinctOptionality, syntax,
-        allowsNullTreatment, allowsSeparator);
+        allowsNullTreatment, allowsSeparator, isPercentile);
+  }
+
+  /** Returns whether this aggregate function is a PERCENTILE function. */
+  public boolean isPercentile() {
+    return isPercentile;
```

Comment:
IMO, PERCENTILE function is a small class of agg functions. So does it deserve a special flag here?
Maybe we can implement a static method like this?

```
public static boolean isPercentile(SqlBasicAggFunction aggFunc) {
    return aggFunc == PERCENTILE_CONT || aggFunc == PERCENTILE_DISC;
}
```

Reply:
Done.